protocol/xmrmaker/claim.go

```diff
@@ -135,41 +202,134 @@ func claimRelayer(
 		return ethcommon.Hash{}, err
 	}
 
-	abi, err := abi.JSON(strings.NewReader(contracts.SwapFactoryMetaData.ABI))
+	calldata, err := getClaimTxCalldata(relayerCommission, contractSwap, secret)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	feeValue, err := calculateRelayerCommission(contractSwap.Value, relayerCommission)
+	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	calldata, err := abi.Pack("claimRelayer", *contractSwap, secret, feeValue)
+	err = waitForClaimReceipt(
+		ctx,
+		ec,
+		txHash,
+		contractAddr,
+		contractSwapID,
+		secret,
+	)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
+	return txHash, nil
+}
+
+func waitForClaimReceipt(
+	ctx context.Context,
+	ec *ethclient.Client,
+	txHash ethcommon.Hash,
+	contractAddr ethcommon.Address,
+	contractSwapID, secret [32]byte,
```

Comment:
Maybe we should start using `gofumpt` to format.  It would have fixed that line.